pkg/prune/prune_test.go

```diff
@@ -0,0 +1,228 @@
+package prune_test
+
+import (
+	"context"
+	"database/sql"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/xmtp/xmtpd/pkg/testutils"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/xmtp/xmtpd/pkg/config"
+	"github.com/xmtp/xmtpd/pkg/db/queries"
+	"github.com/xmtp/xmtpd/pkg/prune"
+)
+
+const (
+	DEFAULT_EXPIRED_CNT = 10
+	DEFAULT_VALID_CNT   = 5
+)
+
+func setupTestData(t *testing.T, db *sql.DB, expired int, valid int) {
+	q := queries.New(db)
+	ctx := context.Background()
+
+	// Insert expired envelopes
+	for i := 0; i < expired; i++ {
+		_, err := q.InsertGatewayEnvelope(ctx, queries.InsertGatewayEnvelopeParams{
+			OriginatorNodeID:     1,
+			OriginatorSequenceID: int64(i),
+			Topic:                []byte("topic"),
+			OriginatorEnvelope:   []byte("payload"),
+			PayerID:              sql.NullInt32{Valid: false},
+			GatewayTime:          time.Now(),
+			Expiry: sql.NullInt64{
+				Int64: time.Now().Add(-1 * time.Hour).Unix(),
+				Valid: true,
+			},
+		})
+		assert.NoError(t, err)
+	}
+
+	// Insert non-expired envelopes
+	for i := 0; i < valid; i++ {
+		_, err := q.InsertGatewayEnvelope(ctx, queries.InsertGatewayEnvelopeParams{
+			OriginatorNodeID:     1,
+			OriginatorSequenceID: int64(i + expired),
+			Topic:                []byte("topic"),
+			OriginatorEnvelope:   []byte("payload"),
+			PayerID:              sql.NullInt32{Valid: false},
+			GatewayTime:          time.Now(),
+			Expiry: sql.NullInt64{
+				Int64: time.Now().Add(1 * time.Hour).Unix(),
+				Valid: true,
+			},
+		})
+		assert.NoError(t, err)
+	}
+}
+
+func makeTestExecutor(
+	t *testing.T,
+	ctx context.Context,
+	db *sql.DB,
+	config *config.PruneConfig,
+) *prune.Executor {
+	return prune.NewPruneExecutor(
+		ctx,
```

Comment:
I've recently discovered `t.Context()`, which is a context that gets canceled at the end of the test. And it stops you from needing to pass around the context separate from `testing.T`.

It's a new one in 1.24

Reply:
oooh! Love it!